src/main/java/io/cryostat/JsonRequestFilter.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright The Cryostat Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.cryostat;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+import jakarta.ws.rs.container.ContainerRequestContext;
+import jakarta.ws.rs.container.ContainerRequestFilter;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.ext.Provider;
+
+@Provider
+public class JsonRequestFilter implements ContainerRequestFilter {
+
+    @Override
+    public void filter(ContainerRequestContext requestContext) throws IOException {
+        if (requestContext.getMediaType() != null
+                && requestContext
+                        .getMediaType()
+                        .isCompatible(jakarta.ws.rs.core.MediaType.APPLICATION_JSON_TYPE)) {
+            BufferedReader reader =
+                    new BufferedReader(new InputStreamReader(requestContext.getEntityStream()));
+            try {
+                String line = reader.readLine();
+                if (line.contains("\"id\"")) {
```

Comment:
I think this check is not quite complete enough. Could you add some simple unit tests to exercise the following cases?

```json
{ "id": 1 }

{
  "id": 1
}

{ "message": "this text includes the string literal \"id\"" }

{
  "foo": "bar",
  "id": 1
}

[
  {
    "id": 1
  }
]
````

It seems to me that this condition only catches the first case where the JSON is a single line, and probably erroneously catches the third case which should be permitted.